share/doc/wake/tutorial.md

```diff
@@ -560,3 +560,129 @@ As a final note, wake files must be UTF-8 encoded.
 This means that you can use Unicode symbols in strings, comments,
 identifiers, and even operators.
 In fact, all the operators we just printed are legal in wake.
+
+## Dealing with failure
+
+Previously, we glossed over functions like `getOrElse` and `getWhenFail`.
+In this section we will explain those as well as the more general concept
+of dealing with operations that can fail in a build flow.
+
+Many operations can fail like taking the first element (`head`) of a `List`
+(it may be empty) or reading a file (it may not exist).
+The simplest and most common way we deal with this in wake is with `Option` as we
+have seen before.
+
+    def myFunction l = match (head l)
```

Comment:
Perhaps it would help with comprehension for some of these code examples to have comments with the type signature of the function? For example, this one could be `# List Int => Int`

Reply:
This seems like the kind of thing that should be consistent throughout the tutorial, but it's not clear to me where it should be introduced. For example, early on we have:
```
    echo 'int helper() { return 42; }' > help.cpp
    git add help.cpp
    cat > tutorial.wake <<'EOF'
    def variant = "native-cpp11-release"
    global def build _ =
      def main = compileC variant ("-I.", Nil) Nil (source "main.cpp")
      def help = compileC variant ("-I.", Nil) Nil (source "help.cpp")
      linkO variant ("-lm", Nil) (main, help, Nil) "tutorial"
    EOF
    wake 'build 0'
```

But I think pointing out that linkO returns `Path` may not be that helpful this early...